### ISSUE TITLE
Wrong sort method implemented in stats/stat_smooth.py

### ISSUE THREAD

**Issue by vdmasek** [yhat/ggpy#607]
When trying to plot a Data Frame with ggplot's stat_smooth there is the following error: 

> 'DataFrame' object has no attribute 'sort'

 Code snippet to reproduce (please don't mind the fact that the dimensions don't make sense):
```
import numpy as np
import pandas as pd
from ggplot import *

# Prepare plot
df_plot = ggplot(aes(x='beef', y="pork"), data=meat) +\
          geom_point() +\
          stat_smooth(colour='blue', span=0.2)
df_plot
```
Looking into it, ggplot v. 0.11.5 is still trying to call sorting methods for pandas versions v. 0.16 or earlier. Currently pandas is on v. 0.21 and has renamed 'sort' -> 'sort_values' since v. 0.17

Solution #1 without considering repercussions of unit tests or any other dependencies:
Change line 77 under `ggplot/stats/stat_smooth.py` as such
`smoothed_data = smoothed_data.sort('x')` becomes: `smoothed_data = smoothed_data.sort_values('x')`

Potentially more backwards compatible solution #2 -> extend the DataFrame from pandas to accept sort()
```
class DataFrameExtended(pandas.DataFrame):
    def sort(self):
        if self:
            return pandas.DataFrame.sort_values(self)
        else:
            return ''

builtins.DataFrame = DataFrameExtended
```

I personally wrote my own class of stat_smooth in a custom.py file and then used the package `builtins` to overwrite the behavior with the code changes.

```
import pandas
import numpy
import builtins
import ggplot

from ggplot.geoms.geom import geom
import ggplot.stats.smoothers
from ggplot.utils import is_date

class stat_smooth(geom):
... changed just line 77 as per solution #1

 # Overwriting the stat_smooth file inside ggplot package in the local env.
builtins.stat_smooth = stat_smooth
```

Cheers!

**Comment by dartdog** [comment]
Thank you for this, I'll also note the following:
from ggplot import *  
ggplot(aes(x='yearbuilt', y='logerror'), data=train_df) + \  
    geom_point(color='steelblue', size=1) + \  
    stat_smooth()  
  ( solution here https://github.com/yhat/ggpy/issues/607)      
yields the following warnings:  
/home/tom/anaconda3/envs/py36/lib/python3.6/site-packages/ggplot/utils.py:81: FutureWarning: pandas.tslib is deprecated and will be removed in a future version.
You can access Timestamp as pandas.Timestamp
  pd.tslib.Timestamp,
/home/tom/anaconda3/envs/py36/lib/python3.6/site-packages/ggplot/stats/smoothers.py:4: FutureWarning: The pandas.lib module is deprecated and will be removed in a future version. These are private functions and can be accessed from pandas._libs.lib instead
  from pandas.lib import Timestamp
/home/tom/anaconda3/envs/py36/lib/python3.6/site-packages/statsmodels/compat/pandas.py:56: FutureWarning: The pandas.core.datetools module is deprecated and will be removed in a future version. Please use the pandas.tseries module instead.
  from pandas.core import datetools

**Comment by vdmasek** [comment]
Thanks for reacting @dartdog! Indeed those warnings are persistent even with commands to restrict them. 

Cheers!

**Comment by kcrisman** [comment]
Yes, I just noticed these `FutureWarnings` just now too - they should be easy to fix.

**Comment by asbhat** [comment]
Looks like the code was already updated 8 months ago to fix this... https://github.com/yhat/ggpy/commit/9d00182343eccca6486beabd256e8c89fb0c59e8

Someone just needs to increment the version and update pypi

**Comment by dgkadesewa** [comment]
@vdmasek 
Hi, thanks. One Question. Did you write this:

_builtins.stat_smooth = stat_smooth_

in the custom.py file or the old stat_smooth.py file?

**Comment by vdmasek** [comment]
Hi @dgkadesewa 
The `builtins.stat_smooth = stat_smooth` is inside my custom.py file. This way I left all the files from the ggplot package intact. Moreover project repository will work on any computer which has the original ggplot package out of the box.
